src/OrleansProviders/Streams/Common/Monitors/DefaultBlockPoolMonitor.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Orleans.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Providers.Streams.Common
+{
+    /// <summary>
```

Comment:
Oh you meant the virtual methods. I leave them virtual because maybe its child classes want to override some of them, for example ,using a different metric name.